components/service/glean/src/test/java/mozilla/components/service/glean/storages/EventsStorageEngineTest.kt

```diff
@@ -350,4 +356,149 @@ class EventsStorageEngineTest {
             ) == snapshot.first().extra)
         assertEquals(1, testGetNumRecordedErrors(testEvent, ErrorType.InvalidValue))
     }
+
+    @Test
+    fun `continue recording events from disk`() {
+        val event = EventMetricType(
+            disabled = false,
+            category = "telemetry",
+            name = "test_event",
+            lifetime = Lifetime.Ping,
+            sendInPings = listOf("store1"),
+            allowedExtraKeys = listOf("foo")
+        )
+
+        // Record the event in the store, without providing optional arguments.
+        EventsStorageEngine.record(
+            metricData = event,
+            monotonicElapsedMs = SystemClock.elapsedRealtime(),
+            extra = mapOf("foo" to "bar")
+        )
+        EventsStorageEngine.testWaitForWrites()
+
+        assertEquals(1, EventsStorageEngine.eventStores["store1"]!!.size)
+        EventsStorageEngine.eventStores.clear()
+
+        // Reinitializing Glean should reload existing events on disk
+        Glean.initialized = false
+        FreshBootDetector.cacheIsFreshBoot = false
+        Glean.initialize(ApplicationProvider.getApplicationContext())
```

Comment:
Does it make sense to slightly change `resetGlean` to allow for skipping data clearing? This would make it future proof, in case something in the way we reset glean in tests changes in the future.

Reply:
That's a good idea.